ui/src/warp_runner/mod.rs

```diff
@@ -70,20 +76,49 @@ impl WarpRunner {
         tokio::spawn(async move {
             // todo: register for events from warp
 
-            let (_account, _messaging, _storage) =
-                match warp_initialization(DEFAULT_PATH.clone(), tesseract.clone(), false).await {
+            let (mut account, mut messaging, _storage) =
+                match warp_initialization(WARP_PATH.clone(), tesseract.clone(), false).await {
                     Ok((i, c, s)) => (i, c, s),
                     Err(_e) => todo!(),
                 };
 
             // this was the only way to get a mutable static variable. but this channel should only be read here.
             let mut rx = rx.lock().await;
 
+            let mut raygun_stream = get_raygun_stream(&mut messaging).await;
+
+            let mut multipass_stream = loop {
```

Comment:
This could be moved into an async function as well to keep it clean.

Reply:
To simplify it, it could probably also be used with the `join!` macro.

Reply:
i like the idea of making separate functions for the futures. but `join!` won't terminate in response to the `Notified` being triggered.